Documentation/tutorial/intro_template.md

```diff
@@ -0,0 +1,170 @@
+
+For example, the following template contains two *Renderer File*s for `conceptual` document type:
+
+```
+/- some_template/
+    |- conceptual.html.primary.tmpl
+    \- concetpual.mta.json.tmpl
+```
+There are two Markdown files `A.md` and `B.md`, the content for `A.md` is:
+
+```markdown
+[Link To B](B.md)
+```
+
+*DocFX Template System* produces two output files for `A.md`: `A.html` and `A.mta.json`, and also two output files for `B.md`: `B.html` and `B.mta.json`. According to `conceptual.html.primary.tmpl`, `.html` is the **primary** output file, the link from `A.md` to `B.md` is resolved to `B.html` instead of `B.mta.json`, which is to say, the content of `A.md` is transformed to:
+
+```html
+<a href="B.html">Link To B</a>
+```
+
+> If no `primary` *Renderer File* is defined, DocFX randomly pick one *Renderer File* as the primary one, and the result is unexpectable.
+
+#### Extended Syntax for Dependencies
+Sometimes a *Renderer File* may have dependencies. For example, a html output file may dependent on some style files. DocFX introduces the following syntax to define the dependency for the *Renderer File*:
+
+```mustache
+{{!include('<file_name>')}}
+```
+
+`docfx` copies these dependencies to output folder preserving its relative path to the *Renderer File*.
+
+> Tips: Mustache is logic-less, and for a sepcific `{{name}}` tag, Mustache searches its context and its parent context recursively. So most of the time [*Preprocessor File*](#-preprocessor-file-) is used to re-define the data model used by the Mustache *Renderer File*.
+
+### *Renderer File* in Liquid syntax
+#### Naming Rule
+*Renderer File*s in [Liquid](https://github.com/Shopify/liquid) syntax **MUST** ends with `.liquid` extension. Liquid contains `include]` tag to support partials, we follow the ruby partials naming convention to have `_<partialName>.liquid` as partial template. 
+
+#### Extended Syntax for Dependencies
+A custom tag `ref`, e.g. `{% ref file1 %}` is introduced to specify the resource files that current template depends on.
+
+*Preprocessor File*
+--------------------
+*Renderer File*s take the data model produced by document processor and render them into output files. Sometimes the data model is not exactly what *Render File*s want. *DocFX Template System* introduces the concept of *Preprocessor* to transform the data model into what *Render File*s exactly want. We call the model *Preprocessor* returns the *View Model*. *View Model* is the model to apply *Renderer File*.
+
+### Naming for *Preprocessor File*
+The naming of *Preprocessor File* follows the naming of *Renderer File* with file extension changes to `.js`: `<renderer_file_name_without_extension>.js`.
+
+If a *Preprocessor File* has no corresponding *Renderer File* however it still needs to be executed, for example, to run [`exports.getOptions` function](#function-signature), it should be named as `<document_type>.tmpl.js`.
```

Comment:
it's the naming rule when *Renderer File* does not exists. We need a naming rule for this case, otherwise style files like main.js will also be considered as *Preprocessor*

---
In reply to: [63292968](https://github.com/dotnet/docfx/pull/306#discussion_r63292968) [](ancestors = 63292968)